java/src/jmri/jmrix/srcp/SRCPThrottle.java

```diff
@@ -167,9 +167,9 @@ void sendUpdate() {
 
         // direction and speed
         msg += (isForward ? " 1" : " 0");
-        msg += " " + ((int) (speedSetting * maxsteps));
+        msg += " " + ((int) (speedSetting * speedStepMode.numSteps));
```

Comment:
This is also building a protocol string.

Reply:
Fixed.